src/cowboy_clear.erl

```diff
@@ -33,15 +33,18 @@ start_link(Ref, Transport, Opts) ->
 
 -spec connection_process(pid(), ranch:ref(), module(), cowboy:opts()) -> ok.
 connection_process(Parent, Ref, Transport, Opts) ->
-	ProxyInfo = case maps:get(proxy_header, Opts, false) of
-		true ->
-			{ok, ProxyInfo0} = ranch:recv_proxy_header(Ref, 1000),
-			ProxyInfo0;
-		false ->
+    ProxyInfo = case maps:get(proxy_header, Opts, false) of
+        true ->
+            case ranch:recv_proxy_header(Ref, 1000) of
+                {ok, ProxyInfo0} -> ProxyInfo0;
+                {error, Reason} -> exit({shutdown, {recv_proxy_header, Reason}});
+                {error, protocol_error, Reason} -> exit({recv_proxy_header, Reason})
```

Comment:
why the differences with/without `shutdown` ?

Reply:
Without shutdown, ranch will print out all the details of the error. see: 
https://github.com/ninenines/ranch/blob/4a1d16242c6fa9865fd7499d301d90b9ca65d6b1/src/ranch_conns_sup.erl#L384

The error printing is suppressed after the `{error, Reason}` clause (where errors are returned from gen_tcp module). But on the `{error, protocol_error, Reason}` clause, these kinds of error reported if parsing the Proxy Protocol failed.

This is the original intention of this implementation.

The clause under {error, Reason} is a bit problematic. But there is no better way to do this at the moment. :(

Reply:
thanks for the clarification.
would be nice to have such info in the code comments.